Towny/src/main/java/com/palmergames/bukkit/towny/tasks/MobRemovalTimerTask.java

```diff
@@ -123,10 +128,19 @@ public void run() {
 					entitiesToRemove.add(entity);
 				}
 				
-				Bukkit.getScheduler().runTask(plugin, () -> {
+				if (entitiesToRemove.isEmpty())
+					return;
+				
+				if (plugin.getScheduler() instanceof FoliaTaskScheduler) {
 					for (LivingEntity entity : entitiesToRemove)
-						entity.remove();
-				});
+						plugin.getScheduler().run(entity, entity::remove);
+				} else {
+					// Remove all mobs in one go if we're on Bukkit
+					plugin.getScheduler().run(() -> {
+						for (LivingEntity entity : entitiesToRemove)
+							entity.remove();
+					});
+				}

```

Comment:
Is this meant to have the implementations split apart? Shouldn't it call the new Scheduler to choose which system is used further down the line?